ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/engine/EmbeddedServer.kt

```diff
@@ -17,18 +17,18 @@ import kotlin.coroutines.*
  * @param TEngine The type of the application engine used by the server.
  * @param TConfiguration The type of the configuration used by the engine.
  */
-public expect class EmbeddedServer<TEngine : ApplicationEngine, TConfiguration : ApplicationEngine.Configuration>(
-    applicationProperties: ApplicationProperties,
-    engineFactory: ApplicationEngineFactory<TEngine, TConfiguration>,
+public expect class EmbeddedServer<TEngine : ServerEngine, TConfiguration : ServerEngine.Configuration>(
+    serverParameters: ServerParameters,
+    engineFactory: ServerEngineFactory<TEngine, TConfiguration>,
     engineConfigBlock: TConfiguration.() -> Unit = {}
 ) {
     /**
      * Provides events on Application lifecycle
      */
     public val monitor: Events
 
-    public val environment: ApplicationEnvironment
-    public val application: Application
+    public val environment: ServerEnvironment
+    public val server: Server
```

Comment:
It feels a bit strange that `EmbeddedServer` has `server` property.
Also, what is difference then between `Server` and `EmbeddedServer` - it's not really clear from the documentation even

Reply:
Yes, it was strange when it had an `application` property, and is now _very_ strange looking.  I think we could get rid of the field entirely, although this would involve some refactoring.